### ISSUE TITLE
1.6.3 performance issues

### ISSUE THREAD

**Issue by skaravad** [tsl0922/ttyd#727]
**Describe the bug**
With latest 1.6.3 we see that ttyd becomes unresponsive , though I don't know what is causing the issue, reverting to 1.6.2 no longer has the problem.

**To Reproduce**
launch ttyd and access it via browser, open multiple sessions with bash as the command 
**Expected behavior**
new sessions should not hang up or become un-responsive.

**Screenshots**

**Environment:**
 - OS: linux debian buster (docker)
 - Browser: Chrome Version 91.0.4472.114 (Official Build) (x86_64)

**Additional context**
Options: `ttyd.x86_64 --port 8000 -P 10 -s 9 --base-path /jumphost -t 'theme={"foreground": "green"}' -t titleFixed="jumphost" -t fontSize=20 -t cursorBlink=true -t cursorStyle=block`

**Comment by tsl0922** [comment]
Can you try if the master version works? ttyd is single threaded, it may hang while waiting process to exit, the waiting logic was moved to thread on master.

**Comment by skaravad** [comment]
@tsl0922 deployed latest master in my env, will monitor for couple of days and will update if we see any issues, so far I was not able to re-produce the issue. 

Thank you.

**Comment by skaravad** [comment]
Found another issue, when we open multiple sessions, if we close one session (browser tab), ttyd exits 
```
[2021/08/31 03:28:19:6069] N:  -- [wsisrv|6|adopted] (3) 11.079s
== uv_read failed with error -5: i/o error
[2021/08/31 03:28:19:6083] N: process killed with signal 1, pid: 13
[2021/08/31 03:28:38:0868] N: WS closed from 172.17.0.1, clients: 2
[2021/08/31 03:28:38:0869] N: killing process, pid: 11
[2021/08/31 03:28:38:0870] N: lws_libuv_closewsi: [wsisrv|5|adopted]
[2021/08/31 03:28:38:0870] N: lws_libuv_closewsi: thr 0: [wsisrv|5|adopted] sa left 2: dyn left: 5 (rk 0)
[2021/08/31 03:28:38:0872] N:  -- [wsisrv|5|adopted] (2) 32.126s
== uv_read failed with error -5: i/o error
[2021/08/31 03:28:38:0879] N: process killed with signal 1, pid: 11
```
It did throw a segfault, but could not find the dump file.

**Comment by tsl0922** [comment]
try the latest release, open a new issue if it still now work for you.
